2011-09-01  Andreas Schwab  <schwab@redhat.com>

	* resolv/res_query.c (__libc_res_nquery): Update assertion.

diff -ru a/resolv/res_query.c b/resolv/res_query.c
--- a/resolv/res_query.c	2011-12-12 14:10:52.000000000 -0700
+++ b/resolv/res_query.c	2011-12-12 14:23:04.832739111 -0700
@@ -248,7 +248,7 @@
 	    && *resplen2 > (int) sizeof (HEADER))
 	  {
 	    /* Special case of partial answer.  */
-	    assert (hp != hp2);
+	    assert (n == 0 || hp != hp2);
 	    hp = hp2;
 	  }
 	else if (answerp2 != NULL && *resplen2 < (int) sizeof (HEADER)
